features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Applications/Create/NewApp.jsx

```diff
@@ -282,22 +284,20 @@ class NewApp extends React.Component {
      * @param {*} appGroups already existing groups
      */
     handleDeleteChip = (chip, index, appGroups) => {
-        this.setState(() => {
-            const { applicationRequest } = this.state;
-            const newRequest = { ...applicationRequest };
-            let values = appGroups || [];
-            values = values.filter(v => v !== chip);
-            newRequest.groups = values;
-            return { applicationRequest: newRequest };
-        });
+        const { applicationRequest } = this.state;
+        const newRequest = { ...applicationRequest };
+        let values = appGroups || [];
+        values = values.filter(v => v !== chip);
+        newRequest.groups = values;
+        this.setState({ applicationRequest: newRequest });
     }
 
     /**
-     * retrieve Settings from the local storage
+     * retrieve Settings from the context and check the application sharing enabled
+     * @param {*} settingsData required data
      */
-    isApplicationGroupSharingEnabled = () => {
-        const settingsData = localStorage.getItem('settings');
-        const enabled = JSON.parse(settingsData).applicationSharingEnabled;
+    isApplicationGroupSharingEnabled = (settingsData) => {
+        const enabled = JSON.parse(settingsData.settings).applicationSharingEnabled;
```

Comment:
I think you can access the context here , No need to pass it as an argument?

Reply:
resolved